### PR TITLE
APPSRE-10279 blockversions for SAPM

### PR DESCRIPTION
**Pull request by fishi0x01**
- SAPM must not open MRs for blocked versions.
- more logging for visibility which gate blocked a promotion

### CODE REVIEW

File: reconcile/saas_auto_promotions_manager/subscriber.py [hunk 1]
```diff
@@ -142,22 +144,42 @@ def _compute_desired_ref(self) -> None:
...
-        ):
-            # We keep current state
-            self.desired_ref = self.ref
-        else:
-            # We have a common single publisher ref w/o any deployment issues
-            self.desired_ref = next(iter(publisher_refs))
+            return
+
+        if not self._passed_accumulated_soak_days():
+            logging.info(
```
Comment (patjlm) [inline reconcile/saas_auto_promotions_manager/subscriber.py:159]: debug

Comment (fishi0x01) [reply reconcile/saas_auto_promotions_manager/subscriber.py]: Good point. Added debug level for soak days

File: reconcile/saas_auto_promotions_manager/subscriber.py [hunk 2]
```diff
@@ -142,22 +144,42 @@ def _compute_desired_ref(self) -> None:
...
+            return
+
+        if any_bad_deployment:
+            logging.info(
+                "Subscriber at path %s promotion stopped because of bad publisher deployment",
+                self.target_file_path,
+            )
+            return
+
+        desired_ref = next(iter(publisher_refs))
```
Comment (hemslo) [inline reconcile/saas_auto_promotions_manager/subscriber.py:172]: in old code, we have `len(publihser_refs) != 1` check, we checked `> 1` case above, can there be any case `publisher_refs` is empty?

Comment (fishi0x01) [reply reconcile/saas_auto_promotions_manager/subscriber.py:172]: Good catch! Added the check :+1:

**Review by hemslo — APPROVED** [review]
LGTM!

**Pull request merged**